Rakefile

```diff
@@ -1239,12 +1260,57 @@ def get_app_prefix (appPath)
     return appPrefix
 end
 
+# Selectively copy files required for the telemetry client
+# based on $telemetryClientInclude and $telemetryClientExclude
+def telemetry_client_copy(fromDir, toDir)
+  # Enumerate and store all included file paths
+  included = []
+  $telemetryClientInclude.each do |path|
+    included.concat(Dir.glob(File.join(fromDir, path)))
+  end
+  
+  # Enumerate and store all excluded file paths
+  excluded = []
+  $telemetryClientExclude.each do |path|
+    excluded.concat(Dir.glob(File.join(fromDir, path)))
+  end
+  
+  # Add .gitignore to excluded files
+  excluded.concat IO.readlines(File.join(fromDir, ".gitignore")).map { |line|
+    line.strip
+  }.select { |line|
+    line.length > 0 && !line.start_with?("#")
+  }.map { |line|
+    path = File.join(fromDir, line)
+  }
+  
+  # Only process the included files without the excluded ones
+  pathExcludedFiles = 0
+  clientFiles = (included-excluded).select { |path| 
+    sw = path.start_with? *excluded
+    pathExcludedFiles += sw ? 1 : 0
+    !sw
+  }
+  
+  # Copy each file to the target directory
+  clientFiles.each do |fromPath|
+    toPath = fromPath.sub(fromDir, toDir)
+    #puts "Copying #{fromPath} to #{toPath}"
+    FileUtils.mkdir_p File.dirname(toPath)
+    FileUtils.cp fromPath, toPath
+  end
+  
+  puts "Copied #{clientFiles.length} Telemetry client files (included #{included.length}, excluded #{excluded.length}, path excluded #{pathExcludedFiles})"
```

Comment:
It just looks scarier than it is I think. It just builds a list of files based on inclusion and exclusion rules and then copies them.